flow/project.py

```diff
@@ -2257,21 +2530,47 @@ def _add_dummy_operation(job):
                 'scheduler_status': JobStatus.dummy}
 
         for job in context['jobs']:
-            has_eligible_ops = any([v['eligible'] for v in job['operations'].values()])
+            # If a job has no operation eligible then we add a dummy operation
+            # Where we show an empty list of operations
+            for a in job['operations'].values():
+                has_eligible_ops = any([b['eligible'] for b in a])
+                if has_eligible_ops:
+                    break
             if not has_eligible_ops and not context['all_ops']:
                 _add_dummy_operation(job)
 
-        op_counter = Counter()
-        for job in context['jobs']:
-            for k, v in job['operations'].items():
-                if k != '' and v['eligible']:
-                    op_counter[k] += 1
-        context['op_counter'] = op_counter.most_common(eligible_jobs_max_lines)
-        n = len(op_counter) - len(context['op_counter'])
+        aggregate_dict = defaultdict(lambda: [])
+        aggregate_counter = Counter()
+        aggregates_per_op = defaultdict(lambda: [])
+
+        for i, job in enumerate(context['jobs']):
+            total_ops = 0
+            for op_name, ops in job['operations'].items():
+                if op_name != '':
+                    for op in ops:
+                        if op['eligible']:
+                            total_ops += 1
+                            agg_id = op['job_ids'].split(' ')
+                            if agg_id not in aggregate_dict[op_name]:
+                                aggregate_dict[op_name].append(agg_id)
+                                if op['is_aggregate']:
+                                    aggregates_per_op[op_name].append(
+                                        (agg_id, op['scheduler_status'])
+                                    )
+            if detailed and compact:
+                context['jobs'][i]['extra_num_operations'] = total_ops - 1
+
+        for op, ags in aggregate_dict.items():
+            aggregate_counter[op] = len(ags)
```

Comment:
why are we doing this in its own for loop?

Reply:
the `aggregate_counter` was used to show how many eligible aggregates are present for an operation. I believe this could be refactored. Although, this will also be tackled in a later PR.